app/Http/Traits/DurV2Helpers.php

```diff
@@ -0,0 +1,575 @@
+                $query->distinct('id');
+            },
+            'applicationPublications' => function ($query) {
+                $query->distinct('id');
+            },
+            'user',
+            'team',
+            'collections' => function ($query) {
+                $query->where('status', Collection::STATUS_ACTIVE);
+            },
+        ])
+        ->when($teamId, function ($query) use ($teamId) {
+            return $query->where(['team_id' => $teamId]);
+        })
+        ->when($status, function ($query) use ($status) {
+            return $query->where(['status' => $status]);
+        })
+        ->first();
+
+        if (!$dur) {
+            throw new NotFoundException();
+        }
+
+        $userDatasets = $dur->userDatasets;
+        $userPublications = $dur->userPublications;
+        $users = $userDatasets->merge($userPublications)
+            ->unique('id');
+        $dur->setRelation('users', $users);
+
+        $applicationDatasets = $dur->applicationDatasets;
+        $applicationPublications = $dur->applicationPublications;
+        $applications = $applicationDatasets->merge($applicationPublications)
+            ->unique('id');
+        $dur->setRelation('applications', $applications);
+
+        unset(
+            $users,
+            $userDatasets,
+            $userPublications,
+            $applications,
+            $applicationDatasets,
+            $applicationPublications,
+            $dur->userDatasets,
+            $dur->userPublications,
+            $dur->applicationDatasets,
+            $dur->applicationPublications
+        );
+
+        // Fetch datasets using the accessor
+        $datasets = $dur->allDatasets  ?? [];
+
+        foreach ($datasets as &$dataset) {
+            $dataset['shortTitle'] = $this->getDatasetTitle($dataset['id']);
+        }
+
+        // Update the relationship with the modified datasets
+        $dur->setAttribute('datasets', $datasets);
+
+        return $dur->toArray();
+    }
+
+    /**
+     * Creates a new tag if it doesn't exist.
+     *
+     * @param mixed $value
+     * @return mixed
+     */
+    public function createNewTagIfNotExists(mixed $value)
+    {
+        if (!is_numeric($value) && !empty($value)) {
+            $tag = Tag::where([
+                'description' => $value,
+            ])->first();
+
+            if (is_null($tag)) {
+                $createdTag = Tag::create([
+                    'type' => TagType::TOPICS,
+                    'description' => $value,
+                    'enabled' => true,
+                ]);
+
+                return $createdTag->id;
+            } else {
+                return $tag->id;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Insert data into ToolHasTag
+     *
+     * @param array $tags
+     * @param integer $toolId
+     * @return mixed
+     */
+    public function insertToolHasTag(array $tags, int $toolId): mixed
```

Comment:
Does this belong in `DurV2Helpers`?  Could be in an equivalent `Tools` file.  Same comment applies for some other functions in this file.

Reply:
Good spot